File: packages/ios/Sources/CivUI/CivMenuItem.swift
// CivUI — CivMenuItem for SwiftUI
// Single entry in a civ-menu. Mirrors civ-menu-item.
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire the platform-native equivalent. See
// .claude/rules/audit-debt.md → "Native platform implementation pass".

import SwiftUI

public struct CivMenuItem: View {
    /// label
    public var label: String

    /// disabled
    public var disabled: Bool

    /// destructive
    public var destructive: Bool

    /// href
    public var href: String

    /// value
    public var value: String

    /// icon
    public var icon: String

    /// Called for analytics tracking (parallels civ-analytics).
    public var onAnalytics: ((String, [String: Any]?) -> Void)?

    public init(
        label: String = "",
        disabled: Bool = false,
        destructive: Bool = false,
        href: String = "",
        value: String = "",
        icon: String = "",
        onAnalytics: ((String, [String: Any]?) -> Void)? = nil
    ) {
        self.label = label
        self.disabled = disabled
        self.destructive = destructive
        self.href = href
        self.value = value
        self.icon = icon
        self.onAnalytics = onAnalytics
    }

    public var body: some View {
        EmptyView()
    }
}

File: packages/android/src/main/kotlin/gov/civui/components/CivMenuItem.kt
// CivUI — CivMenuItem for Jetpack Compose
// Single entry in a civ-menu. Mirrors civ-menu-item.
//
// Placeholder body — the prop surface satisfies schema parity. The
// implementation should wire the Compose-native equivalent. See
// .claude/rules/audit-debt.md → "Native platform implementation pass".

package gov.civui.components

import androidx.compose.foundation.layout.Box
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier

@Composable
fun CivMenuItem(
    label: String = "",
    disabled: Boolean = false,
    destructive: Boolean = false,
    href: String = "",
    value: String = "",
    icon: String = "",
    modifier: Modifier = Modifier,
    onAnalytics: ((event: String, data: Map<String, Any>?) -> Unit)? = null,
    content: @Composable () -> Unit = {}
) {
    Box(modifier = modifier) {
        // Placeholder — see audit-debt.md for native implementation status.
        content()
    }
}
